### PR TITLE
 fix: Accessibility issues for factor dropdown icon

### PR DESCRIPTION
**Pull request by gowthamidommety-okta**
- Adds screen reader text for factor dropdown
- Makes factor dropdown navigable via the keyboard
- Adds aria-expanded and aria-controls to the dropdown

Using MacOS accessibility voice over:

accessibility3

### CODE REVIEW

File: packages/@okta/courage-for-signin-widget/package.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
     "build": "webpack"
   },
   "devDependencies": {
-    "@okta/courage": "4.6.0-beta.2098.gc581fa3",
+    "@okta/courage": "4.6.0-alpha.2340.g2dc3cd2",
```
Comment (haishengwu-okta) [inline packages/@okta/courage-for-signin-widget/package.json]: just an reminder that don't forget to change to beta before merge.

Comment (gowthamidommety-okta) [reply packages/@okta/courage-for-signin-widget/package.json]: yup, will do!

File: src/views/mfa-verify/dropdown/FactorsDropDown.js [hunk 2]
```diff
@@ -28,13 +29,32 @@ function (Okta, FactorsDropDownOptions) {
 
   return BaseDropDown.extend({
     className: 'bg-helper icon-button',
+    screenReaderText: function () {
+      var factors = this.options.appState.get('factors'),
```
[the remark this answers is not in the document]
Comment (gowthamidommety-okta) [reply src/views/mfa-verify/dropdown/FactorsDropDown.js]: 👍

File: src/views/mfa-verify/dropdown/FactorsDropDown.js [hunk 3]
```diff
@@ -30,18 +30,8 @@ function (Okta, FactorsDropDownOptions, Factor) {
...
-        factor = factors.findWhere(_.pick(this.options, 'provider', 'factorType'));
-        if (factor === undefined) {
-          //for factors that support cardinality and only have factorType
-          factor = factors.findWhere(_.pick(this.options, 'factorType'));
-        }
-      } else  {
-        factor = new Factor.Model(this.options.appState.get('factor'), this.toJSON());
-      }
-      factorLabel = factor.get('factorLabel');
+      var factor = FactorUtil.getCurrentFactor(_.extend(this.options, {toJSON: this.toJSON}));
```
Comment (mauriciocastillosilva-okta) [inline src/views/mfa-verify/dropdown/FactorsDropDown.js]: didn't realize in the beginning that we would need to pass the whole JSOn of this dropdown... not sure that is the right thing to do
What about
```
var factors = this.options.appState.get('factors'), factor;
if (factors) {
  factor = FactorUtil.findFactorInFactorsArray(factors, this.options.provider, this.options.factorType);
} else  {
  factor = new Factor.Model(this.options.appState.get('factor'), this.toJSON());
}
factorLabel = factor.get('factorLabel');
```

Comment (gowthamidommety-okta) [reply src/views/mfa-verify/dropdown/FactorsDropDown.js]: we don't need everything.. But we do need appstate, provider and factorType besides toJSON. Should I limit to those options? I like how clear it is when we just use FactorUtil.getCurrentFactor

Comment (gowthamidommety-okta) [reply src/views/mfa-verify/dropdown/FactorsDropDown.js]: Spoke to Mau offline. Changing this

**Pull request merged**